js/msp/MSPHelper.js

```diff
@@ -46,104 +45,95 @@ MspHelper.prototype.process_data = function(dataHandler) {
         case MSPCodes.MSP_RC:
-            RC.active_channels = message_length / 2;
-
+            RC.active_channels = data.byteLength / 2;
             for (var i = 0; i < RC.active_channels; i++) {
-                RC.channels[i] = data.getUint16((i * 2), 1);
+                RC.channels[i] = data.read16();
             }
             break;
         case MSPCodes.MSP_RAW_GPS:
-            GPS_DATA.fix = data.getUint8(0);
-            GPS_DATA.numSat = data.getUint8(1);
-            GPS_DATA.lat = data.getInt32(2, 1);
-            GPS_DATA.lon = data.getInt32(6, 1);
-            GPS_DATA.alt = data.getUint16(10, 1);
-            GPS_DATA.speed = data.getUint16(12, 1);
-            GPS_DATA.ground_course = data.getUint16(14, 1);
+            GPS_DATA.fix = data.read8();
+            GPS_DATA.numSat = data.read8();
+            GPS_DATA.lat = data.readS32();
+            GPS_DATA.lon = data.readS32();
+            GPS_DATA.alt = data.read16();
+            GPS_DATA.speed = data.read16();
+            GPS_DATA.ground_course = data.read16();
             break;
         case MSPCodes.MSP_COMP_GPS:
-            GPS_DATA.distanceToHome = data.getUint16(0, 1);
-            GPS_DATA.directionToHome = data.getUint16(2, 1);
-            GPS_DATA.update = data.getUint8(4);
+            GPS_DATA.distanceToHome = data.read16();
+            GPS_DATA.directionToHome = data.read16();
+            GPS_DATA.update = data.read8();
             break;
         case MSPCodes.MSP_ATTITUDE:
-            SENSOR_DATA.kinematics[0] = data.getInt16(0, 1) / 10.0; // x
-            SENSOR_DATA.kinematics[1] = data.getInt16(2, 1) / 10.0; // y
-            SENSOR_DATA.kinematics[2] = data.getInt16(4, 1); // z
+            SENSOR_DATA.kinematics[0] = data.readS16() / 10.0; // x
+            SENSOR_DATA.kinematics[1] = data.readS16() / 10.0; // y
+            SENSOR_DATA.kinematics[2] = data.readS16(); // z
             break;
         case MSPCodes.MSP_ALTITUDE:
-            SENSOR_DATA.altitude = parseFloat((data.getInt32(0, 1) / 100.0).toFixed(2)); // correct scale factor
+            SENSOR_DATA.altitude = parseFloat((data.readS32() / 100.0).toFixed(2)); // correct scale factor
             break;
         case MSPCodes.MSP_SONAR:
-            SENSOR_DATA.sonar = data.getInt32(0, 1);
+            SENSOR_DATA.sonar = data.readS32();
             break;
         case MSPCodes.MSP_ANALOG:
-            ANALOG.voltage = data.getUint8(0) / 10.0;
-            ANALOG.mAhdrawn = data.getUint16(1, 1);
-            ANALOG.rssi = data.getUint16(3, 1); // 0-1023
-            ANALOG.amperage = data.getInt16(5, 1) / 100; // A
-            ANALOG.last_received_timestamp = Date.now();
+            ANALOG.voltage = data.read8() / 10.0;
+            ANALOG.mAhdrawn = data.read16();
+            ANALOG.rssi = data.read16(); // 0-1023
+            ANALOG.amperage = data.readS16() / 100; // A
+            // this.analog_last_received_timestamp = Date.now(); //TODO, fix this
```

Comment:
I think your rebase has gone wrong here. I changed this to `ANALOG.last_received_timestamp = Date.now()` yesterday for #225, and it probably should stay that way.